openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ParametersGenerator.java

```diff
@@ -141,30 +151,36 @@ public boolean isNullableRequired() {
 
     /**
      * This function for generating parameter ST node for header.
+     * <pre> resource function get pets(@http:Header {name:"x-request-id"} string header) </pre>
      */
-    private RequiredParameterNode handleHeader(Parameter parameter)throws BallerinaOpenApiException {
+    private Node handleHeader(Parameter parameter)throws BallerinaOpenApiException {
         Schema<?> schema = parameter.getSchema();
         TypeDescriptorNode headerTypeName;
         IdentifierToken parameterName = createIdentifierToken(escapeIdentifier(parameter.getName()
                 .toLowerCase(Locale.ENGLISH)), AbstractNodeFactory.createEmptyMinutiaeList(), SINGLE_WS_MINUTIAE);
-        if (schema == null) {
+        if (schema.getType() == null) {
              // Header example:
              // 01.<pre>
              //       in: header
              //       name: X-Request-ID
              //       schema: {}
              //  </pre>
-            return createRequiredParameterNode(createEmptyNodeList(), createIdentifierToken(STRING, SINGLE_WS_MINUTIAE,
-                            SINGLE_WS_MINUTIAE), parameterName);
+            throw new BallerinaOpenApiException(String.format(OAS_SERVICE_106.getDescription(), parameter.getName()));
         } else {
             if (!schema.getType().equals(STRING) && !(schema instanceof ArraySchema)) {
-                //TO-DO: Generate diagnostic about to error type
-                headerTypeName = createBuiltinSimpleNameReferenceNode(null, createIdentifierToken(STRING,
-                        SINGLE_WS_MINUTIAE, SINGLE_WS_MINUTIAE));
+                throw new BallerinaOpenApiException(String.format(OAS_SERVICE_105.getDescription(),
+                        parameter.getName(), schema.getType()));
             } else if (schema instanceof ArraySchema) {
-                String arrayType = ((ArraySchema) schema).getItems().getType();
+                Schema<?> items = ((ArraySchema) schema).getItems();
+                if (items.getType() == null) {
+                    throw new BallerinaOpenApiException(String.format(OAS_SERVICE_104.getDescription(),
+                            parameter.getName()));
```

Comment:
We better move this into an openAPI error creation API, where you can just pass the error enum value and the args instead of repeating this lengthy code everytime we create an error. ( But we don't need to do it now, lets add it to your planed action items :) )